addons/stock/models/stock_location.py

```diff
@@ -184,6 +207,56 @@ def should_bypass_reservation(self):
         self.ensure_one()
         return self.usage in ('supplier', 'customer', 'inventory', 'production') or self.scrap_location or (self.usage == 'transit' and not self.company_id)
 
+    def _check_can_be_used(self, product, quantity=0, package=None):
+        """Check if product/package can be stored in the location. Quantity
+        should in the default uom of product, it's only used when no package is
+        specified."""
+        self.ensure_one()
+        if package and package.package_type_id:
+            return self._check_package_storage(product, package=package)
+        return self._check_product_storage(product, quantity)
+
+    def _check_product_storage(self, product, quantity):
+        """Check if a number of product can be stored in the location. Quantity
+        should in the default uom of product."""
+        self.ensure_one()
+        if self.storage_category_id:
+            # check weight
+            if self.storage_category_id.max_weight < product.weight:
```

Comment:
max_weight should be compared with the sum of weight of product currently in the location + the new product(s)